### PR TITLE
[Yamato] Adding Mac Metal builders

### PR DESCRIPTION
**Pull request by sophiaaar**
### Purpose of this PR
* Adding Mac with Metal to our Yamato testing
* Keeping OpenGLCore, but disabling it from running in CI
* Testing all projects on all platforms will now only happen daily on master, rather than on every commit

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Yamato**: https://yamato.stg.cds.internal.unity3d.com/jobs/45-ScriptableRenderPipeline/tree/macmini-setup

---
### Overall Product Risks
**Technical Risk**: None

**Halo Effect**: None

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: .yamato/upm-ci-universal.yml [hunk 1]
```diff
@@ -4,8 +4,12 @@ platforms:
   - name: Windows64
     type: Unity::VM::GPU
     image: sdet/gamecode_win10:stable
-    flavor: b1.large
+    flavor: m1.large
```
Comment (phi-lira) [inline .yamato/upm-ci-universal.yml]: Is this intentional? Why are we going back to m1?

File: .yamato/upm-ci-universal.yml [hunk 2]
```diff
@@ -4,8 +4,12 @@ platforms:
...
   - name: OSX
+    type: Unity::metal::macmini
+    image: slough-ops/macos-10.14-xcode
+    flavor: m1.mac
```
Comment (phi-lira) [inline .yamato/upm-ci-universal.yml]: Same here

File: .yamato/upm-ci-hdrp.yml [hunk 3]
```diff
@@ -50,18 +50,29 @@ projects:
       
     {% else %}
       
+    - git clone git@github.cds.internal.unity3d.com:unity/utr.git TestProjects/{{ project.folder }}/utr
```
Comment (phi-lira) [inline .yamato/upm-ci-hdrp.yml]: Seems like this git clone is the same for all if/else/if paths. could be out of the loop reducing code duplication/improving maintanability.

File: .yamato/upm-ci-universal.yml (hunk 1 above)
Comment (sophiaaar) [reply .yamato/upm-ci-universal.yml]: not intentional, good spot! thanks!

File: .yamato/upm-ci-universal.yml (hunk 2 above)
Comment (sophiaaar) [reply .yamato/upm-ci-universal.yml]: yes, m1 is all there is for mac (as far as i know) but i will double check

Comment (sophiaaar) [reply .yamato/upm-ci-universal.yml]: confirmed that they need to be m1.mac

File: .yamato/upm-ci-universal.yml (hunk 1 above)
Comment (sophiaaar) [reply .yamato/upm-ci-universal.yml]: fixed

File: .yamato/upm-ci-hdrp.yml (hunk 3 above)
Comment (sophiaaar) [reply .yamato/upm-ci-hdrp.yml]: good shout

Comment (sophiaaar) [reply .yamato/upm-ci-hdrp.yml]: fixed

**Pull request merged**